### ISSUE TITLE
clang 10 issue for endpoint::set_protocol_version implementation

### ISSUE THREAD

**Issue by ropieur** [redboltz/mqtt_cpp#637]
clang 10 complains with
```
include/mqtt/endpoint.hpp:6883:20: error: assigning to 'mqtt::protocol_version' from incompatible type 'std::size_t' (aka 'unsigned long')  
        version_ = version;  
                   ^~~~~~~
```

**Comment by jonesmz** [comment]
If you make a PR that addresses this, it'll certainly be merged.

**Comment by redboltz** [comment]
@ropieur 
Here is the current master code:
https://github.com/redboltz/mqtt_cpp/blob/a710d811d0feeca680d6e141da4d65fe0e74d4b7/include/mqtt/endpoint.hpp#L4004

I think that the issue has already been fixed by #620 .

I close the issue as duplicated.

If you still have problem, reopen the issue please.
